src/ua_timer.c

```diff
@@ -189,9 +225,23 @@ UA_Timer_process(UA_Timer *t, UA_DateTime nowMonotonic,
 
         /* Set the time for the next execution. Prevent an infinite loop by
          * forcing the next processing into the next iteration. */
-        first->nextTime += (UA_Int64)first->interval;
-        if(first->nextTime < nowMonotonic)
-            first->nextTime = nowMonotonic + 1;
+        first->nextTime += (UA_DateTime)first->interval;
+        if(first->nextTime < nowMonotonic) {
+            if(first->handleCycleMissWithBaseTime) {
+                UA_DateTime timeToAdd;
```

Comment:
I see this logic repeated in several places.
Pull it out into a static functions and reuse that.